### PR TITLE
feat(review-workflows): add options in content type builder

### PR DESCRIPTION
**Pull request by nathan-pichon**
### What does it do?

Previously, we were having an "options" object inside the content-type. Still, it was created on content-type creation or update by taking the draftAndPublish attribute (sent at the root level of content-type) and then moving to the options object.
As we do not want CE to be aware of EE features, I had to let the admin panel send top level options attributesthen in BE we move those in the options object. This way, FE avoid a massive refactorization and CE is agnostic from EE features options.

### Why is it needed?

We need to be able to put a new option in content-type called "reviewWorkflows" that is an EE feature, not really a plugin (that's why we didn't store this option under `pluginOptions` object in the content-type.

### How to test it?

Now you can use POST and PUT routes on content-type builder to store a `options` object in the targeted content-type.

E.g.: `PUT http://localhost:1337/content-type-builder/content-types/api::address.address`
```json
{
    "components": [],
    "contentType": {
        "displayName": "Address",
        "singularName": "address",
        "pluralName": "addresses",
        "description": "",
        "draftAndPublish": false,
        "reviewWorkflows": true,
        "pluginOptions": {},
        "kind": "collectionType",
        "collectionName": "addresses",
        "attributes": {
            "postal_code": {
                "type": "string",
                "pluginOptions": {},
                "maxLength": 2
            },
            "categories": {
                "type": "relation",
                "relation": "manyToMany",
                "target": "api::category.category",
                "inversedBy": "addresses",
                "targetAttribute": "addresses",
                "private": false
            },
            "cover": {
                "type": "media",
                "multiple": false,
                "required": false,
                "allowedTypes": [
                    "files",
                    "images",
                    "videos",
                    "audios"
                ],
                "pluginOptions": {}
            },
            "images": {
                "type": "media",
                "multiple": true,
                "required": false,
                "allowedTypes": [
                    "images"
                ],
                "pluginOptions": {}
            },
            "city": {
                "type": "string",
                "required": true,
                "maxLength": 200,
                "pluginOptions": {}
            },
            "json": {
                "type": "json",
                "pluginOptions": {}
            },
            "slug": {
                "type": "uid"
            },
            "notrepeat_req": {
                "type": "component",
                "repeatable": false,
                "pluginOptions": {},
                "component": "blog.test-como",
                "required": true
            },
            "repeat_req": {
                "type": "component",
                "repeatable": true,
                "pluginOptions": {},
                "component": "blog.test-como",
                "required": true
            },
            "repeat_req_min": {
                "type": "component",
                "repeatable": true,
                "pluginOptions": {},
                "component": "blog.test-como",
                "required": false,
                "min": 2
            }
        }
    }
}
```

### Related issue(s)/PR(s)

Let us know if this is related to any issue/pull request

### CODE REVIEW

**Review by gu-stav — COMMENTED** [review]
Two comments from me:

1) The CTB frontend can not handle nesting this attribute in `options` and it would be a huge refactoring I imagine, which is why I'd like to keep it at the top-level as discussed.

2) Draft & publish is not a plugin either, for the sake of consistency I'd keep RW on the same level for now and clean this up later on.

**Comment by nathan-pichon** [comment]
> Two comments from me:
> 
> 1. The CTB frontend can not handle nesting this attribute in `options` and it would be a huge refactoring I imagine, which is why I'd like to keep it at the top-level as discussed.
> 2. Draft & publish is not a plugin either, for the sake of consistency I'd keep RW on the same level for now and clean this up later on.

Oh, I thought it was ok to put an options object. 🤔 
That is problematic because it means that CE should be aware of a "reviewWorkflow" attribute that is not a part of CE. 
Let me think a way to handle this without touching the FE part 👍

**Comment by gu-stav** [comment]
> Oh, I thought it was ok to put an options object. 🤔

For the user-story we agreed to keep it on the top-level I think?

> For the consistency part, D&P is not consistent with itself. When updating / creating a content type, we have the top level attribute, but we store it in options and it doesn't bring any value to have it on top level.

Ok, I can't say anything about that and leave it up to your judgement 👍🏼

**Comment by nathan-pichon** [comment]
> For the user-story we agreed to keep it on the top-level I think?

Yes, I saw that we have made the change on Friday when we discussed the USs. Don't know why I was thinking it didn't change.
Will sort this out in the BE.

File: packages/core/admin/ee/server/register.js [hunk 1]
```diff
@@ -12,6 +13,33 @@ module.exports = async ({ strapi }) => {
...
   }
-
+  if (features.isEnabled('review-workflows')) {
+    addReviewWorkflowMiddleware(strapi);
+  }
   await executeCERegister({ strapi });
 };
+
+/**
+ * A Strapi middleware function that adds support for review workflows.
```
Comment (Marc-Roig) [inline packages/core/admin/ee/server/register.js:23]: Nice idea to use a middleware for this :)

And a minor comment. As a dev looking at this, I would lack info about why we do this.
Wdyt about explaining the frontend can only send the RW at the top level, so we have to move it here?

File: packages/core/admin/ee/server/middlewares/review-workflows.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  contentTypeMiddleware,
+};
+
+/**
+ * A Strapi middleware function that adds support for review workflows.
+ *
+ * Why is it needed ?
+ * For now, the admin panel cannot have anything but top-level attributes in the content-type for options.
+ * But we need the CE part to be agnostics from Review Workflow (which is an EE feature).
+ * CE handle the `options` object, that's why we move the reviewWorkflows boolean to the options object.
```
Comment (Marc-Roig) [inline packages/core/admin/ee/server/middlewares/review-workflows.js:15]: wonderful ✨

**Review by Marc-Roig — APPROVED** [review]

**Pull request merged**